openwisp_controller/geo/api/views.py

```diff
@@ -16,11 +21,25 @@
 Device = load_model('config', 'Device')
 Location = load_model('geo', 'Location')
 DeviceLocation = load_model('geo', 'DeviceLocation')
+FloorPlan = load_model('geo', 'FloorPlan')
 
 
 class DevicePermission(BasePermission):
     def has_object_permission(self, request, view, obj):
-        return request.query_params.get('key') == obj.key
+        if request.query_params.get('token'):
+            received_token = request.query_params.get('token')
+            user_token = request.user.auth_token.key
```

Comment:
I do not know how you made this work but I couldn't, anyway, it's not important, this is not the right path, see my previous comment.